src/index.ts

```diff
@@ -854,6 +870,63 @@ export abstract class PinejsClientCore<PinejsClient> {
 		}
 	}
 
+	private canRetryDefaultHandler(err: any) {
+		const code = err?.statusCode;
+		return code == null || code === 429 || (code >= 500 && code < 600);
+	}
+
+	protected async callWithRetry<T>(
+		fnCall: () => Promise<T>,
+		retry: RetryParameters,
+	): Promise<T> {
+		if (!retry || !this.retry) {
```

Comment:
I think this should be
```suggestion
		if (retry === false || (retry == null && this.retry === false)) {
```
so that if `retry` is explicitly false then it disables retrying and it only looks to the instance level disabling if `retry` is unspecified (right now if it's disabled at instance level it'll always be disabled regardless of per-call setting)

Reply:
If retry parameters are passed in at construction, then yes it will always retry.  If they are not (all legacy code) then it will never retry.  

> If it defaults to an object that means it always retries instead.

No, it would only do so if the `this.retry` was provided, either at construction or later. 

> it needs to be able to distinguish between false/{}/undefined

Adding one more possibility seems unnecessary to to me, but of course I'll go with whatever you think.  

> so that if retry is explicitly false then it disables retrying and it only looks to the instance level disabling if retry is unspecified (right now if it's disabled at instance level it'll always be disabled regardless of per-call setting)

ok, sounds good.